src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,214 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final List<User> correctData = new ArrayList<>();
+    private static final List<User> notValidData = new ArrayList<>();
+    private static final int ONE_USER = 1;
+    private static final int FIRST_USER = 0;
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User testUser;
+
+    @BeforeAll
+    public static void setUp() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl(storageDao);
+        initializeCorrectDataUsers();
+        setNotValidData();
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        testUser = new User();
+    }
+
+    @Test
+    void registerOneUser_OK() {
```

Comment:
it's better to stick to the such test rule for each test case:
given - create test data 
when - call method that you would like to test
then - compare results

It makes tests more readable and avoid some unnecessary logic (or for loop like in your case).